controllers/backup_controller.go

```diff
@@ -142,7 +139,7 @@ func (r *BackupReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 			backup.Status.Phase = apiv1.BackupPhasePending
 			return ctrl.Result{RequeueAfter: 30 * time.Second}, r.Status().Update(ctx, &backup)
 		}
-		backup.Status.SetAsFailed(fmt.Errorf("while getting pod: %w", err))
+		tryFlagBackupAsFailed(ctx, r.Client, &backup, fmt.Errorf("while getting pod: %w", err))
 		r.Recorder.Eventf(&backup, "Warning", "FindingPod", "Error getting target pod: %s",
 			cluster.Status.TargetPrimary)
 		return ctrl.Result{}, r.Status().Update(ctx, &backup)
```

Comment:
I don't think returning an error here makes any sense. Since the backup is marked as failed, it will never rerun a reconciliation cycle. Better to log some meaningful message and `return ctrl.Result{}, nil`